packages/web/src/client/internal/open-feature-client.ts

```diff
@@ -181,6 +182,21 @@ export class OpenFeatureClient implements Client {
     return this.evaluate<T>(flagKey, this._provider.resolveObjectEvaluation, defaultValue, 'object', options);
   }
 
+  track(occurrenceKey: string, occurrenceDetails: TrackingEventDetails): void {
+
+    this.shortCircuitIfNotReady();
+
+    const context = {
```

Comment:
It may be worth adding a comment explaining why you're creating a new context object. I'm assuming it's because you don't want the provider to be able to modify context in the track method.

Reply:
I did this and also froze the context.